### ISSUE TITLE
Use Protobuf builders instead of Validating Builders

### ISSUE THREAD

**Issue by armiol** [SpineEventEngine/core-java#1039]
Currently, the framework users are obligated to use the validating builders while working with entity states. Such an approach features a fail-fast validation upon a setter invocation:

```java
@Apply
void on(TaskCreated event) {
    builder().setName(event.getName());
}
```
In this case, if the `name` field of the state is required, and `event.getName()` is empty, the end-user receives an exception raised in the setter. So it is possible to surround this block with `try ... catch` and handle the error. Without a `try ... catch` block, the diagnostic is still acceptable, as the stack trace is going to point at the setter line.

However, in case of some complex validation rules, such an approach does not work as conveniently. Let's say we have a message, that is a part of some entity state:

```proto
message PersonName {
    option (required_field) = "given_name|honorific_prefix & family_name";

    string honorific_prefix = 1;
    string given_name = 2;
    string middle_name = 3;
    string family_name = 4;
    string honorific_suffix = 5;
}
```

Validating such a complex rule, say, in the `Aggregate`'s event applier is too early, as the properties of this object may be set in a batch of events and hence — in another applier. So instead of fail-fast approach, any constraint violations are discovered during the transaction commit and out of an end-user's control.

Another thing is that the validating builders aren't a graceful extension of Protobuf's own `Message` builders. The API of `Message` builders provides more features, and `ValidatingBuilder` codebase is always downstream — as the new versions of Protobuf Java library are released from time to time.

To deal with these issues we will perform the actions as follows:

1. Extend the codebase of Protobuf `Message` builders (assuming `M` is a type of the message):
    * generate `public M vbuild() throws ValidationException` method, which validates and builds the message;
    * generate `public void validate() throws ValidationException` method, which performs its validation.
2. In scope of entity `Transaction` — if the entity state validation fails, emit the system `EntityValidationFailed` event, consisting of
    * Entity ID.
    * Entity state type.
    * ID and type of the last message, which caused the issue.
    * The ID of the root message (e.g. a `Command` instance), which was the first in a chain.
    * The validation error itself.
  We may post the event from the `orError(..)` method of a respective `Repository` — as the event bus is already available there.
3. Migrate the entity contract from using `ValidatingBuilder` to using `Message.Builder` corresponding to the entity state.

The expected benefits are
* no fail-fast validation — as it works inconsistently;
* the end-users will be able to take advantage of native Protobuf Java API of builders. 
* the end-users will still be able to trigger the validation in `try ... catch` block, but do that explicitly.
* the clients will be able to subscribe to `EntityValidationFailed` event — as they know the ID of the root message, such as `Command` — and handle it accordingly.

**Comment by armiol** [comment]
@alexander-yevsyukov @dmdashenkov FYI.

**Comment by armiol** [comment]
Implemented in #1068 and [base#409](https://github.com/SpineEventEngine/base/pull/409).
